### PR TITLE
fix(opentelemetry): fix timestamps on otel spans

### PR DESCRIPTION
**Pull request by mabdinur**
Timestamps in OpenTelemetry Spans are integers and have a precession of nanoseconds ([reference](
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#timestamp), [code](https://github.com/open-telemetry/opentelemetry-python/blob/v1.0.0/opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py#L741)).

- In `Span.__init__(self, ...., start_time, ...)` start_time should be an integer with a precision of nanoseconds.
- In `Span.end(end_time)` end_time should be an integer with a precision of nanoseconds.
  

Opentelemetry support is unannounced so a release note is not required.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] PR description includes explicit acknowledgement/acceptance of the performance implications of this PR as reported in the benchmarks PR comment.

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

File: ddtrace/_opentelemetry/span.py [hunk 1]
```diff
@@ -39,8 +39,8 @@ def __init__(
     ):
         # type: (...) -> None
         if start_time is not None:
-            # otel instrumentation tracks time in nanoseconds while ddtrace uses seconds
-            datadog_span.start = start_time / 1e9
+            # start_time should be set in nanoseconds
+            datadog_span.start_ns = start_time
```
Comment (mabdinur) [inline ddtrace/_opentelemetry/span.py:44]: This is just a refactor.

File: ddtrace/_opentelemetry/span.py [hunk 2]
```diff
@@ -84,9 +85,9 @@ def end(self, end_time=None):
 
         :param end_time: The end time of the span, in nanoseconds. Defaults to ``now``.
         """
-        # end_time set in nanoseconds while Datadog finish_time is set in seconds
-        dd_finish_time = end_time if end_time is None else end_time / 1e9
-        self._ddspan.finish(dd_finish_time)
+        if end_time is None:
+            end_time = time_ns()
+        self._ddspan._finish_ns(end_time)
```
Comment (mabdinur) [inline ddtrace/_opentelemetry/span.py:90]: This is the fix. end_time should be nanoseconds (previously it was seconds)

File: tests/opentelemetry/test_span.py [hunk 3]
```diff
@@ -107,6 +107,21 @@ def test_otel_span_is_recording(oteltracer):
     assert span.is_recording() is False
 
 
+def test_otel_span_end(oteltracer):
+    start_time_ns = 1680522337 * 1e9
+    duration_sec = 11
+    end_time_ns = start_time_ns + duration_sec * 1e9
```
Comment (ZStriker19) [inline tests/opentelemetry/test_span.py:113]:
```suggestion
    end_time_ns = start_time_ns + (duration_sec * 1e9)
```
for clarity?

Comment (mabdinur) [reply tests/opentelemetry/test_span.py:113]: I don't think not a big deal

**Comment by mabdinur** [comment]
> ## Benchmarks
> Comparing candidate commit [2fe66c7](https://github.com/DataDog/dd-trace-py/commit/2fe66c792ee2a2382f094f8f408d91f27d44ec01) in PR branch `munir/otel-fix-timestamps` with baseline commit [cdbebfb](https://github.com/DataDog/dd-trace-py/commit/cdbebfb96340b1a3c1dd20d8ec128a024fb7790e) in branch `1.x`.
> 
> Found 0 performance improvements and 1 performance regressions! Performance is the same for 55 cases.
> 
> #### scenario:span-start-finish
> * 🟥 `execution_time` [+0.062ms; +0.094ms] or [+2.081%; +3.164%]

According to this benchmark, this PR adds 30 microseconds to `Span.finish()`. This overhead was introduced when `Span.finish()` was refactored into two functions: `Span.finish()` and `Span.finish_ns()`. This benchmark finishes Spans [without registering any span finish callbacks](https://github.com/DataDog/dd-trace-py/blob/v1.10.0/benchmarks/span/scenario.py#L33) (ex: SpanAggregator, TopLevelProcessor, etc) so it does not provide an accurate duration of ending a ddtrace span. When the expected usage of `ddtrace.Span()` is taken into account this 30 microsecond overhead add <1% overhead and should be acceptable.

**Pull request merged**